packages/babel-generator/src/source-map.ts

```diff
@@ -58,7 +58,7 @@ export default class SourceMap {
           setSourceContent(
             map,
             resolvedSources[i],
-            this._inputMap.sourcesContent[i],
+            this._inputMap.sourcesContent?.[i],
```

Comment:
Nit: We can skip this full loop if there's no `sourcesContent`.

Reply:
Does this affect `sources`? At first I skipped the whole loop, but then I'm not sure if `sources` will be preserved.

Reply:
Instead of eagerly adding all sources, it'll only adds source as they are marked. It's not important to keep a source that was input if that source is never seen in the map, but setting a `sourcesContent` forces it.

Reply:
Thanks!